central/declarativeconfig/manager_impl.go

```diff
@@ -151,11 +184,27 @@ func (m *managerImpl) runReconciliation() {
 }
 
 func (m *managerImpl) reconcileTransformedMessages(transformedMessagesByHandler map[string]protoMessagesByType) {
-	for handler, protoMessagesByType := range transformedMessagesByHandler {
+	log.Debugf("Run reconciliation for the next handlers: %v", maputil.Keys(transformedMessagesByHandler))
+	transformedMessages := map[reflect.Type][]proto.Message{}
+	for _, protoMessagesByType := range transformedMessagesByHandler {
 		for protoType, protoMessages := range protoMessagesByType {
-			// TODO(ROX-14693): Add upserting transformed resources.
-			log.Debugf("Upserting transformed messages of type %s from file %s: %+v",
-				protoType.Name(), handler, protoMessages)
+			transformedMessages[protoType] = append(transformedMessages[protoType], protoMessages...)
+		}
+
+	}
+	for _, protoType := range protoTypesOrder {
+		messages, hasMessages := transformedMessages[protoType]
+		if !hasMessages {
+			continue
+		}
+		typeUpdater, hasUpdater := m.updaters[protoType]
+		if !hasUpdater {
+			log.Fatalf("Manager does not have updater for type %v", protoType)
```

Comment:
Since the customer is not in control of which proto types can be YAML configurations be transformed to and which proto types can be updated, the lack of corresponding updater is an error of us writing incorrect code. I think it is basically an incorrect state of the system, and perhaps we should stop entire `manager_impl` if we see such a type. That's an argument against sending this error down the `ProcessError` pipe.

The argument for sending this error is that `System Health` currently is the only page where this error will be visible to customers. Perhaps the flow should be:
1. If we encounter a such error, we send it to `ProcessError` with the message `FATAL: Manager cannot process proto type {proto type}. Processing declarative configurations will stop now. Please contact support to investigate the issue`
2. After that we immediately stop `manager_impl` from running